servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpSetCookie.java

```diff
@@ -277,7 +295,9 @@ static HttpSetCookie parseSetCookie(final CharSequence setCookieString, boolean
             }
         }
 
-        return new DefaultHttpSetCookie(name, value, path, domain, expires, maxAge, isWrapped, isSecure, isHttpOnly);
+        assert name != null && value != null; // these are checked at runtime in the constructor
```

Comment:
why you even need this if its checked in the constructor ?

Reply:
IntelliJ flags this as a warning (e.g. passing in arguments that maybe null that are no annotated with `@Nullable`).